app/src/components/market/sections/market_pooling/market_pool_liquidity_container.tsx

```diff
@@ -14,10 +15,19 @@ interface Props {
 }
 
 const MarketPoolLiquidityContainer: React.FC<Props> = (props: Props) => {
-  const { isScalar } = props
+  const {
+    isScalar,
+    marketMakerData: { address },
+  } = props
 
-  if (isScalar) return <ScalarMarketPoolLiquidity {...props} />
-  return <MarketPoolLiquidity {...props} />
+  const history = useHistory()
+
+  useEffect(() => {
+    history.replace(`/${address}/pool`)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+
+  return <>{isScalar ? <ScalarMarketPoolLiquidity {...props} /> : <MarketPoolLiquidity {...props} />}</>
```

Comment:
I'm wondering if every instance of this could have been done more DRY. For example in the top component where we change currentTab and where we setCurrentTab so that we can maybe use useEffect on tab change and change url based on it? Have you tried that approach? That way we can also maybe avoid hardcoding everything and it would be easier to maintain?

Reply:
Nope I didn't, I will try! Thank you